### ISSUE TITLE
Style/RedundantParentheses breaks pattern matching

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#14239]
It wants to remove parens that are needed to keep precedence right.

## Expected behavior
```ruby
lhs = {foo: 1}
rhs = {bar: 2}
if (lhs in {foo:}) \
   && (rhs in {bar:})
  puts 32
else
  puts 23
end
```

```
$ ruby rubocop/redundant_parens.rb
32
```

```
$ bundle exec rubocop -A rubocop/redundant_parens.rb
Inspecting 1 file
F

Offenses:

rubocop/redundant_parens.rb:1:1: C: [Corrected] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
lhs = {foo: 1}
^
rubocop/redundant_parens.rb:3:4: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a one-line pattern matching.
if (lhs in {foo:}) \
   ^^^^^^^^^^^^^^^
rubocop/redundant_parens.rb:4:7: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a one-line pattern matching.
   && (rhs in {bar:})
      ^^^^^^^^^^^^^^^
rubocop/redundant_parens.rb:5:4: F: Lint/Syntax: unexpected token tANDOP
(Using Ruby 3.3 parser; configure using TargetRubyVersion parameter, under AllCops)
   && rhs in {bar:}
   ^^
rubocop/redundant_parens.rb:7:1: F: Lint/Syntax: unexpected token kELSE
(Using Ruby 3.3 parser; configure using TargetRubyVersion parameter, under AllCops)
else
^^^^

1 file inspected, 5 offenses detected, 3 offenses corrected
```
```
$ ruby rubocop/redundant_parens.rb
rubocop/redundant_parens.rb: --> rubocop/redundant_parens.rb
expected `then` or `;` or '\n'cannot parse the expression
> 4  if lhs in {foo:} \
> 5     && rhs in {bar:}
> 6    puts 32
> 7  else
> 8    puts 23
> 9  end

rubocop/redundant_parens.rb:5: syntax error, unexpected &&, expecting `then' or ';' or '\n' (SyntaxError)
   && rhs in {bar:}
   ^~

```

## RuboCop version

```
$ bundle exec rubocop -V
1.76.0 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.45.0, analyzing as Ruby 3.3, running on ruby 3.3.8) [arm64-darwin24]
  - rubocop-minitest 0.38.1
  - rubocop-performance 1.25.0
```

**Comment by bkuhlmann** [comment]
Hey. :wave: This doesn't appear to be quite fixed yet as I'm still seeing the following with RuboCop 1.76.1:

``` 
lib/marameters/probe.rb:60:31: C: [Correctable] Style/RedundantParentheses: Don't use parentheses around a one-line pattern matching.
    def only_double_splats? = (parameters in [[:keyrest]] | [[:keyrest, *]])
```

If I remove the parenthesis, I'll get a syntax error.

**Comment by koic** [comment]
@bkuhlmann I don't see any syntax errors caused by the removal of parentheses. If this issue reproduces, can you open a new issue based on the template?

```console
$ echo 'def only_double_splats? = (parameters in [[:keyrest]] | [[:keyrest, *]])' | RUBOCOP_TARGET_RUBY_VERSION=3.0 bundle exec rubocop --stdin foo.rb --only Style/RedundantParentheses -a
Inspecting 1 file
C

Offenses:

foo.rb:1:27: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a one-line pattern matching.
def only_double_splats? = (parameters in [[:keyrest]] | [[:keyrest, *]])
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
====================
def only_double_splats? = parameters in [[:keyrest]] | [[:keyrest, *]]
```
